### DISCUSSION TITLE
Boost CLI enhancement feedback

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/boost#576] [Storage Provider]
## Overview
Recent feedback in pre-release testing and over the slack channels state that SPs would like to see more CLI commands in boost. The purpose of this discussion is to collect details on what CLI commands they need in boost and the motivation behind their request.

## Feedback Form

- Full CLI command (Ex: boostd dagstore initialize-all)
   Motivation behind the command (If there is already a UI option for this then we need to understand why CLI is required as well)

**Comment by brendalee** [comment]
@stuberman would be great to get your specific input here!

**Reply by scaseye** [reply]
stop, restart/resume, retry from the CLI would be great.

**Comment by stuberman** [comment]
Sorry, since this request came out I had a motherboard failure and then once fixed, the version of boost I upgraded to v1.1.0-rc1 caused my boost node to be unusable for almost a week. Now have upgraded to v1.16.0-rc3 and v1.1.0-rc2 and things are working again.

Commands I miss:
```
lotus-miner storage-deals pending-publish
lotus-miner storage-deals pending-publish --publish-now
lotus-miner --call-on-markets stop
```
Other commands - not so sure if there is still a way to do an offline deal with a remote client (over the wire)
Also would like to run all Evergreen commands on Boost node including retrieve:

`lotus client retrieve --provider f01278 --maxPrice 0 --allow-local --car '[DataCID]' $(pwd)/xxx__yyy.car`

Today I have to retrieve using lotus-miner and then copy the downloaded file to boost node.

**Comment by dirkmc** [comment]
```
lotus-miner storage-deals pending-publish
lotus-miner storage-deals pending-publish --publish-now
```
These commands have equivalents in the web UI under the Publish Deals page. It shows a list of deals waiting to be published and there is a `Publish Now` button.

<img width="771" alt="Screen Shot 2022-06-27 at 9 29 42 AM" src="https://user-images.githubusercontent.com/169124/175884162-377a63e3-4baf-4f89-be3b-4f29a3c4165b.png">

If you want to invoke the same commands from the command line you can do so with a curl:

```
$ curl -H "Content-Type: application/json" \
  -d '{"query":"mutation { dealPublishNow }"}' \
  http://localhost:8080/graphql/query

{"data":{"dealPublishNow":true}}
```


> not so sure if there is still a way to do an offline deal with a remote client

The client can make an offline deal with `boost offline-deal`
The SP imports the deal with `boostd import-data`

> Also would like to run all Evergreen commands on Boost node including retrieve

Currently retrieval works the same way in Boost as it did in lotus, so you need to use the lotus client.
This should work from the machine that is running boost, you just need to export the `FULLNODE_API_INFO` environment variable and point it at a lotus full node (lotus daemon).
